### PR TITLE
Feature/message processing

### PR DESCRIPTION
**Pull request by IvanovAndrew**

### CODE REVIEW

File: Bang/Game/DrawChecker.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+            bool draw = ShouldApplyEffect(firstCard);
+            
+            gameplay.Discard(firstCard);
+
+            if (victim is LuckyDuke)
+            {
+                var secondCard = gameplay.DealCard();
+
+                draw |= ShouldApplyEffect(secondCard);
```
Comment (bashnia007) [inline Bang/Game/DrawChecker.cs:22]: I am not a big fun of bitwise operators. Do you really neead it here?

File: Bang/Game/GamePlayInitializer.cs [hunk 2]
```diff
@@ -1,32 +1,51 @@
...
-
-
-namespace Domain.Game
+using Bang.Characters;
+using Bang.Exceptions;
+using Bang.PlayingCards;
+using Bang.Roles;
+using Bang.Game;
+using Bang.Players;
+using Gameplay;
```
Comment (bashnia007) [inline Bang/Game/GamePlayInitializer.cs:12]: **Minor:** Not used usings

File: Bang/GameEvents/ReplyActionMessage.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
+using System.Xml.Schema;
```
Comment (bashnia007) [inline Bang/GameEvents/ReplyActionMessage.cs:1]: unused using

File: Bang/PlayingCards/Visitors/ICardTypeVisitor.cs [hunk 4]
```diff
@@ -0,0 +1,97 @@
+using Bang.GameEvents.CardEffects;
+using Bang.Weapons;
+
+namespace Bang.PlayingCards.Visitors
```
Comment (bashnia007) [inline Bang/PlayingCards/Visitors/ICardTypeVisitor.cs:4]: Why all these interfaces are in one file?

File: Server.Tests/GameTests.cs [hunk 5]
```diff
@@ -1,12 +1,9 @@
-﻿using Domain.Game;
```
[the remark this answers is not in the document]
Comment (IvanovAndrew) [reply Server.Tests/GameTests.cs:1]: Fixed

File: Bang/PlayingCards/BangGameCard.cs [hunk 6]
```diff
@@ -1,37 +1,43 @@
...
+        public string Description => Type.Description;
+        
+        public BangGameCard(CardType card, Suite suite, Rank rank)
+        {
+            Type = card ?? throw new ArgumentNullException(nameof(card));
+            Suite = suite;
+            Rank = rank;
+        }
+
+        public bool IsLongTerm => Type.Accept(new IsLongTermCardTypeVisitor());
```
[the remark this answers is not in the document]
Comment (IvanovAndrew) [reply Bang/PlayingCards/BangGameCard.cs]: Fixed

File: Bang/PlayingCards/BangGameCard.cs [hunk 7]
```diff
@@ -1,37 +1,43 @@
...
+        }
+
+        protected override int GetHashCodeCore()
+        {
+            return HashCode.Combine(Type, Suite, Rank);
+        }
+        
+        public static implicit operator CardType(BangGameCard card)
+        {
+            return card?.Type;
```
[the remark this answers is not in the document]
Comment (IvanovAndrew) [reply Bang/PlayingCards/BangGameCard.cs]: Fixed

File: Bang/Game/DrawChecker.cs (hunk 1 above)
Comment (IvanovAndrew) [reply Bang/Game/DrawChecker.cs]: I think it's the same as 
draw = draw || ShoudApplyEffect(secondCard)

But you're right: [the logical conditional operators don't support compound assignment](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/boolean-logical-operators).

Fixed

File: Bang/Game/GamePlayInitializer.cs (hunk 2 above)
Comment (IvanovAndrew) [reply Bang/Game/GamePlayInitializer.cs]: Fixed

File: Bang/GameEvents/ReplyActionMessage.cs (hunk 3 above)
Comment (IvanovAndrew) [reply Bang/GameEvents/ReplyActionMessage.cs]: Fixed

File: Bang/PlayingCards/Visitors/ICardTypeVisitor.cs (hunk 4 above)
Comment (IvanovAndrew) [reply Bang/PlayingCards/Visitors/ICardTypeVisitor.cs:3]: Fixed

**Pull request merged**